internal/agent/device/lifecycle/manager.go

```diff
@@ -92,15 +96,71 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 }
 
 func (m *LifecycleManager) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	// this controller currently does not implement a sync operation
 	return nil
 }
 
 func (m *LifecycleManager) AfterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	if current.Decommission == nil && desired.Decommission != nil {
+		m.log.Warnf("Detected decommissioning request from flightctl service")
+		errored := false
+		m.updateWithStartedCondition(ctx)
+		m.log.Warnf("Updated Condition to decommissioning started")
+
+		// todo - switch handling based on decommission target type once we support more types;
+		// keep track of whether we have an error
+
+		if !errored {
+			m.log.Warnf("No errors during decommissioning, updating Condition to decommissioning completed")
+			m.updateWithCompletedCondition(ctx)
+		}
+
+		// no return
+		m.log.Warnf("Preparing to wipe agent certificate and keys and reboot")
+		m.wipeAndReboot()
```

Comment:
since this will return an error its existence should send error status or what is the expected result

Reply:
I can bubble up the error